### ISSUE TITLE
./gradlew check fails on an M1 Mac because there is no arm64 variant of the node version used

### ISSUE THREAD

**Issue by sniezek** [geb/issues#678]
Hi! When I run `./gradlew check` on the master branch of [Geb](https://github.com/geb/geb), I get the following error:
```
Could not determine the dependencies of task ':doc:site:nodeSetup'.
> Failed to query the value of task ':doc:site:nodeSetup' property 'nodeArchiveFile'.
   > Could not resolve all files for configuration ':doc:site:detachedConfiguration21'.
      > Could not find org.nodejs:node:8.17.0.
        Searched in the following locations:
          - https://repo.maven.apache.org/maven2/org/nodejs/node/8.17.0/node-8.17.0.pom
          - https://oss.sonatype.org/content/repositories/snapshots/org/nodejs/node/8.17.0/node-8.17.0.pom
          - https://oss.sonatype.org/content/repositories/releases/org/nodejs/node/8.17.0/node-8.17.0.pom
          - https://nodejs.org/dist/v8.17.0/node-v8.17.0-darwin-arm64.tar.gz
        Required by:
            project :doc:site
```
Any help will be greatly appreciated!

**Comment by erdi** [comment]
Looks like you are building on an M1 and the build is using a very outdated version of node to process some frontend resources for the site for which there is no arm64 variant. To fix it I would need to upgrade quite a lot of stuff and it's not going to be on my priority list for a while. 

May I ask you why you are trying to build the project? Are you attempting to contribute?

**Comment by sniezek** [comment]
Thank you for a quick response, I understand. I was trying to build the project because I wanted to build it with Gradle Enterprise and to contribute any findings.

**Comment by erdi** [comment]
I see. Would using `./gradlew -p module check` for now give you enough to play with? Or is being able to run all of the check tasks across the whole project essential to what you are doing? FWIW, I believe everything apart from `doc:site` should be buildable on M1.

Also, are you doing this just for yourself or are you planning to contribute anything back? Or are you doing some research for Gradle Enterprise purposes? Reading your comment again I believe you are trying to use Gradle Enterprise to find issues/inefficiencies in the build and then contribute fixes back, is that right?

**Comment by erdi** [comment]
Maybe I should get my act together and finally setup publishing of build scans from each build on CI...

**Comment by sniezek** [comment]
Sorry for my message being not clear, but you got it right! I have just joined Gradle and will be working on their Gradle Enterprise product. A part of the onboarding process is to get familiar with GE by using it with an open source project and to try to analyze and optimize the build, and to contribute any optimizations that won't require adding GE to the project. Due to not being able to build Geb, I chose a different open source project to work on. But I really like Geb so maybe I will try to optimize that one in the future too!
